File: ThematicMapCreator.Client/src/app/models/layer-style-options/graduated-colors-style-options.ts
import { LayerStyleOptions } from './layer-style-options';
import { LayerStyle } from './layer-style.enum';

export class GraduatedColorsStyleOptions implements LayerStyleOptions {
    style: LayerStyle.GraduatedColors;
    propertyName: string;
    size = 8;
    minColor = '#000000';
    maxColor = '#FFFFFF';
    minValue: number;
    maxValue: number;
}

File: ThematicMapCreator.Api/Contracts/LayerStyleOptions/GraduatedColorsStyleOptions.cs
using System;

namespace ThematicMapCreator.Api.Contracts.LayerStyleOptions
{
    public class GraduatedColorsStyleOptions : ILayerStyleOptions
    {
        public string MaxColor { get; set; }
        public double MaxValue { get; set; }
        public string MinColor { get; set; }
        public double MinValue { get; set; }
        public string PropertyName { get; set; }
        public double Size { get; set; }
        public LayerStyle Style { get; } = LayerStyle.GraduatedColors;
    }
}
